modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/connectivity/oauth/UpdatingAuthorizationCodeState.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.runtime.connectivity.oauth;
+
+import static org.mule.runtime.module.extension.internal.runtime.connectivity.oauth.ExtensionsOAuthUtils.toAuthorizationCodeState;
+import org.mule.runtime.extension.api.connectivity.oauth.AuthorizationCodeState;
+import org.mule.runtime.oauth.api.AuthorizationCodeOAuthDancer;
+import org.mule.runtime.oauth.api.builder.AuthorizationCodeListener;
+import org.mule.runtime.oauth.api.state.ResourceOwnerOAuthContext;
+
+import java.util.Optional;
+import java.util.function.Consumer;
+
+/**
+ * An implementation of {@link AuthorizationCodeListener} which registers an {@link AuthorizationCodeListener}
+ * in order to get updated state when a refresh token operation is completed or the resource is simply re-authorized.
+ * 
+ */
+public class UpdatingAuthorizationCodeState implements AuthorizationCodeState {
+
+  private AuthorizationCodeState delegate;
+  private final Consumer<ResourceOwnerOAuthContext> onUpdate;
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this unused "onUpdate" private field. ([squid:S1068](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS1068))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7744&issues=AWmiFXZHotIXnmsUruQ4&open=AWmiFXZHotIXnmsUruQ4)

[ISSUE_KEY]: AWmiFXZHotIXnmsUruQ4